src/main/java/org/scijava/io/DataHandle.java

```diff
@@ -82,12 +82,31 @@
 	void setLength(long length) throws IOException;
 
 	/**
-	 * Verifies that the handle has sufficient bytes available to read, returning
-	 * the actual number of bytes which will be possible to read, which might
-	 * be less than the requested value.
+	 * Gets the number of bytes which can be safely read from, or written to, the
+	 * data handle, bounded by the specified number of bytes.
+	 * <p>
+	 * In the case of reading, attempting to read the returned number of bytes is
+	 * guaranteed not to throw {@link EOFException}. However, be aware that the
+	 * following methods <em>might still process fewer bytes</em> than indicated
+	 * by this method:
+	 * </p>
+	 * <ul>
+	 * <li>{@link #read(ByteBuffer)}</li>
+	 * <li>{@link #read(ByteBuffer, int)}</li>
+	 * <li>{@link #read(byte[])}</li>
+	 * <li>{@link #read(byte[], int, int)}</li>
+	 * <li>{@link #skip(long)}</li>
+	 * <li>{@link #skipBytes(int)}</li>
+	 * </ul>
+	 * <p>
+	 * In the case of writing, attempting to write the returned number of bytes is
+	 * guaranteed not to expand the length of the handle; i.e., the write will
+	 * only overwrite bytes already within the handle's bounds.
+	 * </p>
 	 * 
-	 * @param count Number of bytes to read.
-	 * @return The actual number of bytes available to be read.
+	 * @param count Desired number of bytes to read/write.
+	 * @return The actual number of bytes which could be safely read/written,
```

Comment:
I'd remove "safely" here. It's not about safety, it's about feasibility. You cannot read past the end of the data.